core/includes/form.inc

```diff
@@ -4594,7 +4596,13 @@ function form_validate_number(&$element, &$form_state) {
     $offset = isset($element['#min']) ? $element['#min'] : 0.0;
 
     if (!valid_number_step($value, $element['#step'], $offset)) {
-      form_error($element, t('%name is not a valid number.', array('%name' => $name)));
+      // Integer steps are the most common use case, so give a specific message.
+      if ($element['#step'] == 1 && is_numeric($offset) && $offset == round($offset)) {
+        form_error($element, t('%name: the value must be an integer.', array('%name' => $name)));
+      }
+      else {
+        form_error($element, t('%name: the value must be an integral number of steps %step from %offset.', array('%name' => $name, '%step' => $element['#step'], '%offset' => $offset)));
```

Comment:
I'm not so sure the average person would catch what "integral" means here? That word commonly used has a different meaning from integer. I had to Google it to confirm the meaning myself.

Reply:
Good point. I'm so used to the mathematical interpretation that I forget about the common-language meaning. I'll update the wording.

Reply:
"A whole number of steps."

Reply:
Yup, I agree. "whole number" makes more sense than "integral number" if you're not a mathematician 👍🏼